common/src/main/java/bio/terra/catalog/service/DatasetService.java

```diff
@@ -170,10 +175,13 @@ public void deleteMetadata(DatasetId datasetId) {
   public String getMetadata(DatasetId datasetId) {
     var dataset = datasetDao.retrieve(datasetId);
     ensureActionPermission(dataset, SamAction.READ_ANY_METADATA);
-    return new DatasetResponse(
-            dataset, getService(dataset.storageSystem()).getDataset(dataset.storageSourceId()))
-        .convertToObject()
-        .toString();
+    StorageSystemInformation information;
+    try {
+      information = getService(dataset.storageSystem()).getDataset(dataset.storageSourceId());
+    } catch (ForbiddenException e) {
```

Comment:
The correctness of `ForbiddenException` depends on the TDR API continuing to work the same way in the future. If it's changed to a 404 / `NotFoundException` then the code would break. So maybe it'd be better to catch the more general `ErrorReportException` here instead. Or, add an integration test that makes sure that TDR is working as this code expects it to.

Reply:
It turns out that this was the wrong exception and needed fixing. It was easy to add an integration test, so I've done that as well.